### PR TITLE
docs: Add docstring for torch.masked._ops.logaddexp

### LINKED ISSUE #113082: Add Docstrings to ./torch/_refs/__init__.py, ./torch/_refs/special/__init__.py, ./torch/masked/_ops.py
Add docstrings on the following lines. Follow these [docstring guidelines](https://github.com/pytorch/pytorch/blob/main/CONTRIBUTING.md#docstring-type-formatting). A docstring should have:
* A short one-sentence description in imperative mode.
* A longer description.
* A list of associated Args.
* A return value (for functions)
* An example
Follow the existing patterns in the file you are editing. After adding the docstring, run:
```
lintrunner -a
```
- **File**: `./torch/_refs/special/__init__.py`, **Line**: 231
- **File**: `./torch/masked/_ops.py`, **Line**: 1479
- **File**: `./torch/_refs/__init__.py`, **Line**: 1080
- **File**: `./torch/_refs/__init__.py`, **Line**: 1090
- **File**: `./torch/_refs/__init__.py`, **Line**: 1098
- **File**: `./torch/_refs/__init__.py`, **Line**: 1106

cc @carljparker

### PR DESCRIPTION
**Pull request by bjhargrave**
logaddexp is not a reduction and normalization, so
_apply_docstring_templates cannot be used to add a docstring.

Fixes https://github.com/pytorch/pytorch/issues/113082

Also fix another misspelling.

cc @svekars @carljparker

### CODE REVIEW

File: torch/masked/_ops.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
     DimOrDims = Optional[Tuple[int]]
 
 
-__all__ = []
+__all__ = ["logaddexp"]  # Cannot use _apply_docstring_templates
```
Comment (kit1980) [inline torch/masked/_ops.py:24]: No need to add to all, it's available as `torch.masked.logaddexp`.

File: torch/masked/_ops.py [hunk 2]
```diff
@@ -1486,6 +1485,48 @@ def logaddexp(
     input_mask: Optional[Tensor] = None,
     other_mask: Optional[Tensor] = None,
 ) -> Tensor:
+    """logaddexp(input, other, *, dtype=None, input_mask=None, other_mask=None) -> Tensor
+
+Returns logaddexp of all the elements in the :attr:`input` and the :attr:`other`
+tensor. The :attr:`input` elements are masked out according to the boolean tensor
+:attr:`input_mask` and the attr:`other` elements are masked out according to the boolean tensor
+:attr:`other_mask`. For the logaddexp operation, the value for masked out elements is ``-2147483648``.
```
Comment (kit1980) [inline torch/masked/_ops.py:1493]: Is -2147483648 is true for logaddexp? (looks like it taken from logsumexp docstring).

**Comment by kit1980** [comment]
@cpuhrsch Could you take a look if this makes sense?

**Review by cpuhrsch — APPROVED** [review]

File: torch/masked/_ops.py (hunk 2 above)
Comment (cpuhrsch) [reply torch/masked/_ops.py:1493]: I'm sorry, but "For the logaddexp operation, the value for masked out elements is ``-2147483648``" doesn't really make sense. Could we please elaborate or remove this?

Either an element is masked out, which means it has a position, but no value, or it is not. A masked out element cannot have a value.

**Review by cpuhrsch — CHANGES_REQUESTED** [review]

File: torch/masked/_ops.py (hunk 2 above)
Comment (bjhargrave) [reply torch/masked/_ops.py:1493]: > (looks like it taken from logsumexp docstring)

It is. 

The real answer for both logsumexp and logaddexp is the smallest value which the tensor's dtype can hold. `-torch.inf` for float types and `torch.iinfo.min` for integral.

https://github.com/pytorch/pytorch/blob/498a76080299d44a3e22787bc42ec8e2eba71905/torch/masked/_ops.py#L421-L425

**Comment by bjhargrave** [comment]
@cpuhrsch Given the implementation, a masked out value and changed to the smallest value which the tensor's dtype can hold. `-torch.inf` for float types and `torch.iinfo.min` for integral. This is the same for logsumexp (since it reuses that logic).

https://github.com/pytorch/pytorch/blob/498a76080299d44a3e22787bc42ec8e2eba71905/torch/masked/_ops.py#L421-L425

**Comment by cpuhrsch** [comment]
@bjhargrave - Yes, but that should be an implementation detail. The reduction identity is used to implement the masked operation.

**Comment by bjhargrave** [comment]
> Yes, but that should be an implementation detail. The reduction identity is used to implement the masked operation.

I removed the sentence.

**Review by cpuhrsch — APPROVED** [review]
Thank you for sending the fix!

**Comment by svekars** [comment]
Still need to fix lint?

**Comment by kit1980** [comment]
> Still need to fix lint?

Looks unrelated.

**Comment by bjhargrave** [comment]
> Looks unrelated.

Yes, I looked at the failure and it seemed to be in some test code.

**Pull request closed without merging**